documentation/source/_templates/page.html

```diff
@@ -0,0 +1,225 @@
+{# Copied from https://github.com/pradyunsg/furo/blob/2021.11.23/src/furo/theme/furo/page.html to extend it with the GitHub button.
+ # When furo gets support for this natively (https://github.com/pradyunsg/furo/discussions/114) remove this file.
+ #}
+{% extends "base.html" %}
+
+{% block body -%}
+{% include "partials/icons.html" %}
+{% from "partials/edit-this-page.html" import determine_page_edit_link with context %}
+
+  <style>
+/* this was copied from '.theme-toggle' in furo's admonitions.sass.
+   It causes the button's background to be invisible.
+
+   Ideally, this CSS would be unified with that but it can't currently
+   be because there's a javascript event that gets attached to .theme-toggle.
+ */
+.btn-github {
+  background: transparent;
+  border: none;
+  cursor: pointer;
+  padding: 0;
+}
+
+.btn-github svg {
+  color: var(--color-foreground-primary);
+  width: 1em;
+  height: 1em;
+}
+
+body[data-theme="*"] .btn-github svg {
+  display: block;
+}
+
+.mobile-header .edit-button-container {
+  float: left;
+  padding: 1rem .5rem;
+}
+
+@media (max-width: 67em) {
+  .content-icon-container .edit-button-container {
+    display: none;
+  }
+}
+  </style>
+
+<input type="checkbox" class="sidebar-toggle" name="__navigation" id="__navigation">
+<input type="checkbox" class="sidebar-toggle" name="__toc" id="__toc">
+<label class="overlay sidebar-overlay" for="__navigation">
+  <div class="visually-hidden">Hide navigation sidebar</div>
+</label>
+<label class="overlay toc-overlay" for="__toc">
+  <div class="visually-hidden">Hide table of contents sidebar</div>
+</label>
+
+{% if theme_announcement -%}
+<div class="announcement">
+  <aside class="announcement-content">
+    {% block announcement %} {{ theme_announcement }} {% endblock announcement %}
+  </aside>
+</div>
+{%- endif %}
+
+<div class="page">
+  <header class="mobile-header">
+    <div class="header-left">
+      <label class="nav-overlay-icon" for="__navigation">
+        <div class="visually-hidden">Toggle site navigation sidebar</div>
+        <i class="icon"><svg><use href="#svg-menu"></use></svg></i>
+      </label>
+    </div>
+    <div class="header-center">
+      <a href="{{ pathto(master_doc) }}"><div class="brand">{{ docstitle if docstitle else project }}</div></a>
+    </div>
+    <div class="header-right">
+     <div class="edit-button-container">
+        <a class="edit-button" href="{{ determine_page_edit_link() }}">
+          <button type="button" class="btn-github" title="{{ _('Edit this page') }}">
+            <svg><use href="#svg-github"></use></svg>
+          </button>
+        </a>
+      </div>
```

Comment:
I looked over the [changes you made](https://www.diffchecker.com/8M5dOMRn) and the core idea here seems fine!

Just to make sure I'm understand correctly, though: 

* Internally the furo theme has its own `_templates` directory containing the original copies of all of these files, right? 
* And, by having our own `_templates` directory, we're overriding the default templates (to include the "Edit on GitHub" changes you made)?

Reply:
One suggestion: To make these changes easier to understand in the future, maybe one thing we could do is add HTML comments before/after to make it apparent which blocks are ours, and which blocks come from Furo:

```suggestion
     <!-- start of changes added by @kousu in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3624  -->
     <div class="edit-button-container">
        <a class="edit-button" href="{{ determine_page_edit_link() }}">
          <button type="button" class="btn-github" title="{{ _('Edit this page') }}">
            <svg><use href="#svg-github"></use></svg>
          </button>
        </a>
      </div>
      <!-- end of changes added by @kousu in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/3624  -->
```

That way, a dev looking back on this work doesn't have to diff with the original Furo file to see what the changes were.

Reply:
I started out by pasting in furo's page.html, then adding a commit that replaced the buttons with `{{ include 'partials/buttons.html' }`, then a third commit that added in the patch I wanted to see. So, in that case, you could easily tell what had happened. But I was defeated having to redo the same for the mobile header icons -- they're not quite able to share the same partial because of how the CSS is set up.

As an aside: in this case, `sphinx-book-theme` scores a few points over `furo`: it makes heavier use of partials and makes customizations a lot easier, and easier to track. I wish `furo` did too; I think it will eventually because furo's author is hard at work on componentizing all themes in https://github.com/pradyunsg/sphinx-basic-ng/tree/main/src/sphinx_basic_ng/theme/basic-ng/components it's just not been backported there yet.

I could redo the current version to have a 'import from furo' commit followed by a 'patch in the github button' commit. Would that be satisfying?

Reply:
At the same time, if we're anticipating this becoming native in furo, this might not be worth the effort anyway.

Reply:
> I could redo the current version to have a 'import from furo' commit followed by a 'patch in the github button' commit. Would that be satisfying?

I think that would work! But, up to you, really. It's a bit of a nitpick... and it's possible that the next time we return to this will be to just remove it anyway. (Once it becomes native in furo)

Reply:
Ah we have squashes forced on this repo so I couldn't do it that way afterall